src/api/app/models/bs_request.rb

```diff
@@ -725,73 +740,35 @@ def change_review_state(new_review_state, opts = {})
       unless new_review_state.in?([:new, :accepted, :declined, :superseded])
         raise InvalidStateError, "review state must be new, accepted, declined or superseded, was #{new_review_state}"
       end
-      found = false
 
       old_request_state = state
+      review = find_review_for_opts(opts)
+      raise Review::NotFoundError unless review
 
-      reviews_seen = {}
-      reviews.reverse_each do |review|
-        matching = true
-        matching = false if review.by_user && review.by_user != opts[:by_user]
-        matching = false if review.by_group && review.by_group != opts[:by_group]
-        matching = false if review.by_project && review.by_project != opts[:by_project]
-        matching = false if review.by_package && review.by_package != opts[:by_package]
-
-        rkey = "#{review.by_user}@#{review.by_group}@#{review.by_project}@#{review.by_package}"
-
-        # This is needed for MeeGo BOSS, which adds multiple reviews by_user
-        # FIXME3.0: think about review ordering and make reviews addressable
-        next unless matching && !(reviews_seen.key?(rkey) && review.state == :accepted)
-        reviews_seen[rkey] = 1
-        found = true
-        comment = opts[:comment] || ''
-        next unless review.state != new_review_state || review.reviewer != User.current.login || review.reason != comment
-        review.reason = comment
-        review.state = new_review_state
-        review.reviewer = User.current.login
-        review.save!
-        Event::ReviewChanged.create(notify_parameters)
-
-        history = nil
-        history = HistoryElement::ReviewAccepted if new_review_state == :accepted
-        history = HistoryElement::ReviewDeclined if new_review_state == :declined
-        history = HistoryElement::ReviewReopened if new_review_state == :new
-        history.create(review: review, comment: opts[:comment], user_id: User.current.id) if history
-      end
```

Comment:
It took me quite a while to map all the removed code to it's new location. I hope I did not overlook any... Next time please split such changes into smaller chunks, aka. commits.
And some more details in the commit message would help.